src/main/java/org/apache/cassandra/sidecar/utils/ChecksumVerifier.java

```diff
@@ -30,9 +31,19 @@
 public interface ChecksumVerifier
 {
     /**
-     * @param checksum  expected checksum value
-     * @param filePath  path to SSTable component
-     * @return String   component path, if verification is a success, else a failed future is returned
+     * @param options  a map with options required for validation
+     * @param filePath path to SSTable component
+     * @return a future String with the component path if verification is a success, otherwise a failed future
      */
-    Future<String> verify(String checksum, String filePath);
+    Future<String> verify(MultiMap options, String filePath);
```

Comment:
The checksum verifier should not be responsible for parsing headers for the same reason mentioned in the other comment. 

I think you are missing a server-side `Digest` data structure, which supplies the input for verification. 

One approach is this.

```java
public interface ChecksumVerifier<D extends Digest>
{
    Future<String> verify(D digest, String filePath);

    default boolean canVerify(D digest)
    {
        return false;
    }
}

// And then,
public class MD5ChecksumVerifier extends AsyncFileChecksumVerifier<MD5Digest>
```

Beside that, as a nit, I would rename `ChecksumVerifier` to `DigestVerifier` on server code for consistency and accuracy.